controller_manager:
  ros__parameters:
    update_rate: 30  # Hz
    use_sim_time: false

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    diffbot_base_controller:
      type: diff_drive_controller/DiffDriveController

diffbot_base_controller:
  ros__parameters:
    left_wheel_names: ["left_wheel_joint"]
    right_wheel_names: ["right_wheel_joint"]

    wheel_separation: 0.45
    wheel_radius: 0.1

    # Odometry
    publish_rate: 50.0
    odom_frame_id: "odom"
    base_frame_id: "base_link"
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.001]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.001]
    open_loop: true
    enable_odom_tf: true

    # Velocity and acceleration limits
    linear:
      x:
        has_velocity_limits: true
        max_velocity: 0.5  # m/s
        has_acceleration_limits: true
        max_acceleration: 1.0  # m/s^2
    angular:
      z:
        has_velocity_limits: true
        max_velocity: 1.0  # rad/s
        has_acceleration_limits: true
        max_acceleration: 2.0  # rad/s^2
